src/parser.R

```diff
@@ -1,145 +1,180 @@
-library(stringr)
+library(stringr)   # For text manipulation
 library(testthat)  # For unittesting: test_that
-library(here) # For relative working directory. Looks for where there is a file with .Rproj extension and sets that folder as the root. 
-
-
-# Parsing the txt files into a dataframe
-# using the library stringr and regex to extract the information 
-# from the text files and turn it into a dataframe
-#
-# this function looks at a multiple files, breaks it down by month, day, 
-# parses it and then binds everything together to form one dataset
-crawlerResultPath <- file.path(getwd(),                       # gets the working root directory  
-                               'crawler_result_Conversion',   # folder where we want to save the PDFs and 
-                               fsep = .Platform$file.sep)     # makes sure the path is platform agnostic
-
-
-# List of RegEx used
-listOfTxtFiles = list.files(crawlerResultPath, pattern = ".txt")
-dayStartRegEx <- "^For Date:\\s*(\\d{2}\\/\\d{2}\\/\\d{4})\\s*-\\s*(\\w*)"
-callRegEx <- "^(\\s|\\d{2}-\\d*)\\s*(\\d{4})\\s*(.*?)\\s{2,}(.*)"
-locationRegEx <- "^\\s*Location/Address:\\s*(.*)"
-officeRegEx <- "^\\s*ID:\\s*(.*)"
-callTakerRegEx <- "^\\s*Call Taker:\\s*(.*)"
-dispClrdRegEx <- "^\\s*Disp-(\\d{2}:\\d{2}:\\d{2})\\s*(Arvd-\\d{2}:\\d{2}:\\d{2})?\\s*Clrd-(\\d{2}:\\d{2}:\\d{2})"
-wasArrestMadeRegEx <- "^\\s*Refer To Arrest:\\s*(.*)"
-referToSummonRegEx <- "^\\s*Refer To Summons:\\s*(.*)$"
-summonRegEx <- "^\\s*Summons:\\s*(.*)$"
-arrestRegEx <- "^\\s*Arrest:\\s*(.*)$"
-addressRegEx <- "^\\s*Address:\\s*(.*)$"
-ageRegEx <- "^\\s*Arrest:\\s*(.*)$"
-chargesRegEx <- "^\\s*Charges:\\s*(.*)$"
-
-parseTxtToDf <- function(filePath){
+library(pdftools)  # For PDF to Text Conversion
+
+# TODO(Jorge3fFernandes): Use Apply family to get rif of for loops
+
+crawlerResultPath <- "./crawler_result_Conversion" # text and pdf location
+
+listAllPDFs <- list.files(crawlerResultPath, 
+                          pattern = ".pdf", 
+                          full.names = TRUE )
+
+oneText <- function(pdfList){
```

Comment:
Why would you want to append all the pdf's to a single text file?

I don't see how this will be beneficial especially when you're planning on using `apply` functions to vectorize the parsing.

Reply:
That way I can go line by line and identify where the day started and where the call started. Another reason is because sometimes they upload the data only for one day, sometime for a whole month and I just want to be careful by just collapsing everything.

I see your point and I'll try to see if there is a more efficient way to parse it when I start rebuilding the functions.

Reply:
perhaps you can make a decision to just collapsing monthly reports into a single file rather than all the files.